### ISSUE TITLE
mosaic=0, but mosaic still work

### ISSUE THREAD

**Issue by WoWoGG** [THU-MIG/yolov10#464]
following is my args.yaml:

task: detect
mode: train
model: ultralytics/cfg/models/v10/yolov10n.yaml
data: D:/pythonProject4/datasets/data.yaml
epochs: 100
time: null
patience: 150
batch: 4
imgsz: 2000
save: true
save_period: -1
val_period: 1
cache: false
device: 0
workers: 4
project: run/temp2
name: exp
exist_ok: false
pretrained: true
optimizer: SGD
verbose: true
seed: 0
deterministic: false
single_cls: false
rect: false
cos_lr: false
close_mosaic: 0
resume: false
amp: false
fraction: 1.0
profile: false
freeze: null
multi_scale: false
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: true
split: val
save_json: false
save_hybrid: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
vid_stride: 1
stream_buffer: false
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
embed: null
show: false
save_frames: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
show_boxes: true
line_width: null
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: null
workspace: 4
nms: false
lr0: 0.001
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
label_smoothing: 0.0
nbs: 4
hsv_h: 0
hsv_s: 0
hsv_v: 0
degrees: 180
translate: 0
scale: 0
shear: 0
perspective: 0.0
flipud: 0.5
fliplr: 0.5
bgr: 0.0
mosaic: 0
mixup: 0
copy_paste: 0.3
auto_augment: randaugment
erasing: 0.1
crop_fraction: 1.0
cfg: null
tracker: botsort.yaml
save_dir: run\temp2\exp

![Uploading train_batch0.jpg…]()

**Comment by WoWoGG** [comment]
![Uploading train_batch0.jpg…]()

**Comment by WoWoGG** [comment]
![train_batch2](https://github.com/user-attachments/assets/29ba94db-983b-43dd-a45b-0299567ce814)
